docs/fundamentals/code-analysis/quality-rules/il3002.md

```diff
@@ -0,0 +1,20 @@
+---
+title: "IL3002: Avoid calling members annotated with 'RequiresAssemblyFilesAttribute' when publishing as a single file (code analysis)"
+description: "Members annotated with 'RequiresAssemblyFilesAttribute' require assembly files to be on disk"
+ms.date: 15/04/2020
+ms.topic: reference
+author: mateoatr
+ms.author: matorre
+f1_keywords:
+  - "IL3002"
+  - "RequiresAssemblyFiles"
+---
+# IL3002: Avoid calling members annotated with 'RequiresAssemblyFilesAttribute' when publishing as a single file.
+
+## Cause
+
+When publishing as a single-file (for example, by setting the PublishSingleFile property in a project to true), calling members annotated with the `RequiresAssemblyFilesAttribute` attribute might be unsafe, given that members annotated with this attribute require assembly files to be on disk and the assemblies embedded in a single-file are memory loaded.
+
+## How to fix violations
+
+Consider removing the violating call.
```

Comment:
The message displayed from the attribute should contain helpful info - so we should mention that user should do both - try to remove the call, or adapt the code according to the message.